src/App.tsx

```diff
@@ -1,109 +1,75 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+import { ItemPerPage } from './types/ItemPerPage';
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
+const itemsCount: ItemPerPage[] = [3, 5, 10, 20];
+
 export const App: React.FC = () => {
+  const [selectedPage, setSelectedPage] = useState(1);
+  const [itemsPerPage, setItemsPerPage] = useState<ItemPerPage>(5);
+
+  const startItem = itemsPerPage * selectedPage - itemsPerPage;
+  const finishItem =
+    itemsPerPage * selectedPage > items.length
+      ? items.length
+      : itemsPerPage * selectedPage;
+
+  const preparedItem = items.slice(startItem, finishItem);
+
+  const handleSetItemsPerPage = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setItemsPerPage(+event.target.value as ItemPerPage);
+    setSelectedPage(1);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {selectedPage} (items {startItem + 1} - {finishItem} of{' '}
+        {items.length})
```

Comment:
we can make it more readable by using interpolation
```suggestion
        {`Page ${selectedPage} (items ${startItem + 1} - ${finishItem} of ${items.length})`}
```